<!-- 
         ROS navigation system
-->
<launch>
  <!-- Velocity smoother nodelet; we must provide the absolute name of nodelet manager, as we can be in an application namespace -->
  <node pkg="nodelet" type="nodelet" name="navigation_velocity_smoother" output="screen" respawn="true"
        args="load yocs_velocity_smoother/VelocitySmootherNodelet /mobile_base_nodelet_manager">
    <rosparam file="$(find thorp_navigation)/param/vel_smoother.yaml" command="load"/>
    <remap from="navigation_velocity_smoother/smooth_cmd_vel" to="/cmd_vel_mux/input/navigation"/>
    <remap from="navigation_velocity_smoother/robot_cmd_vel" to="/mobile_base/commands/velocity"/>
    <remap from="navigation_velocity_smoother/odometry" to="/odom"/>
  </node>

  <!-- Move base -->
  <node pkg="move_base" type="move_base" name="move_base" output="screen" respawn="true">
    <rosparam file="$(find thorp_navigation)/param/move_base/costmap_common_params.yaml" command="load" ns="global_costmap"/>
    <rosparam file="$(find thorp_navigation)/param/move_base/costmap_common_params.yaml" command="load" ns="local_costmap"/>
    <rosparam file="$(find thorp_navigation)/param/move_base/local_costmap_params.yaml"  command="load"/>
    <rosparam file="$(find thorp_navigation)/param/move_base/global_costmap_params.yaml" command="load"/>

    <rosparam file="$(find thorp_navigation)/param/move_base/base_local_planner_params.yaml" command="load"/>
    <rosparam file="$(find thorp_navigation)/param/move_base/move_base_params.yaml" command="load"/>

    <remap from="bumpers" to="/mobile_base/sensors/bumper_pointcloud"/>
    <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/>
    <remap from="odom" to="/odom"/>
  </node>
</launch>
